sds_data_manager/utils/stackbuilder.py

```diff
@@ -51,65 +44,60 @@ def build_sds(
     # other step function processing steps
     processing_step_function = step_function_stack.ProcessingStepFunctionStack(
         scope,
-        f"ProcessingStepFunctionStack-{sds_id}",
-        sds_id,
+        "ProcessingStepFunctionStack",
         dynamodb_table_name=dynamodb.table_name,
         env=env,
     )
 
     data_manager = sds_data_manager_stack.SdsDataManager(
         scope,
-        f"SdsDataManager-{sds_id}",
-        sds_id,
+        "SdsDataManager",
         open_search,
         dynamodb,
         processing_step_function_arn=processing_step_function.sfn.state_machine_arn,
         env=env,
     )
 
-    domain = domain_stack.Domain(
-        scope,
-        f"DomainStack-{sds_id}",
-        sds_id,
-        env=env,
-        use_custom_domain=use_custom_domain,
-    )
+    domain = None
+    domain_name = account_config.get("domain_name", None)
+    if domain_name is not None:
+        domain = domain_stack.DomainStack(
+            scope,
+            "DomainStack",
+            domain_name=domain_name,
+            account_name=account_config["account_name"],
+            env=env,
+        )
 
     api_gateway_stack.ApiGateway(
         scope,
-        f"ApiGateway-{sds_id}",
-        sds_id,
+        "ApiGateway",
         data_manager.lambda_functions,
+        domain_stack=domain,
         env=env,
-        hosted_zone=domain.hosted_zone,
-        certificate=domain.certificate,
-        use_custom_domain=use_custom_domain,
     )
 
-    networking = networking_stack.NetworkingStack(
-        scope, f"Networking-{sds_id}", sds_id, env=env
-    )
+    networking = networking_stack.NetworkingStack(scope, "Networking", env=env)
 
     instrument_list = ["Codice"]  # etc
 
-    lambda_code_directory = Path(__file__).parent / ".." / "lambda_code" / "SDSCode"
+    lambda_code_directory = Path(__file__).parent.parent / "lambda_code" / "SDSCode"
     lambda_code_directory_str = str(lambda_code_directory.resolve())
 
     for instrument in instrument_list:
         ecr = ecr_stack.EcrStack(
             scope,
-            f"{instrument}Processing-{sds_id}",
```

Comment:
Is this something we want to share? Don't we want to have a distinguishing name here?

Reply:
I figure each instrument only wants to create a single ECR stack for now, the images within those registries should be tagged differently though. I think this is something we should discuss about how to prototype as a team though without squashing people's work. We can add a configuration parameter to the `cdk.json` file somewhere with a unique id to pass around... But the problem is if I'm not developing on this specific piece I don't want that unique id to be passed around everywhere, only the specific piece I'm working on. So I figured it was easier to just manually while I develop go in and add my own "sds-id" to the stack I'm working on if I need it.

Other thoughts are welcome on this!

Reply:
I suppose I can see it both ways. If we're expecting to regularly deploy large parts of the stack with unique names, then it would be really annoying to go through and add an sds-id to everything.

 On the other hand, if you are only deploying one or two things, then it's fine. I'm concerned that excessively sharing resources will cause problems, but at least for our development, I don't think that many people are working on the CDK code at the same time... and hopefully it's decoupled enough that we could share buckets and everything. 

I'm not sure what the best answer  is or what my preference is, so I guess defaulting to removing it and seeing if it causes problems makes sense to make the code cleaner.